### ISSUE TITLE
Warning for overlapping geometries?

### ISSUE THREAD

**Issue by antonszilasi** [mostaphaRoudsari/Honeybee#573]
Hi Guys,

I am a bit confused by what is going on here, when I was running a workshop there was a case where there were 2 overlapping geometries and EnergyPlus threw an error (unfortunately I can't remember which one) I've tried to reproduce the error with 3 over lapping geometry's here.

![overlapping geometries](https://cloud.githubusercontent.com/assets/8773483/20021226/ee2eef16-a2a2-11e6-8610-02552f88489c.jpg)

EnergyPlus runs fine, despite the fact that there are 3 intersecting masses, two of which are invalid breps. What exactly is EnergyPlus doing here? I can't imagine how 3 zones can occupy the same space?

Furthermore if I don't use the intersect masses component it still runs despite the fact that the zones are intersecting! 

Do you think that we should put a quality check here?

**Comment by antonszilasi** [comment]
@mostaphaRoudsari  I also wanted to confirm what you said the other day which was that EnergyPlus doesn't need a closed zone in order to perform a calculation, is that correct? Does that mean that it only assumes heat transfer through the actual surfaces that are present?

**Comment by chriswmackey** [comment]
@antonszilasi 

EnergyPlus, like most simulation engines, is only as intelligent as it's input.  It can take a whole range of things and run a calculation but it won't always tell you everything that is wrong with the model.  I get the sense that the E+ development team thought that basic geometry checks like this are things that should be handled by either the user or the interface (rather than the engine).  However, intersection calculations are very computationally intensive and including these in Honeybee would really increase the runtime and memory usage for large models.  I feel that our general rule of not penalizing the experts who build their model well should hold in this case.  Let me know if you disagree or if you have a clever way of checking the intersections with a quick calculation that does not penalize those who build their models well.


And, yes, E+ does not need a closed zone to run.  This is just something that we included in Honeybee for our own sanity, best practice and (in a more personal case), to help with zone air volume calculations performed by the comfort maps.

-Chris

**Comment by TheodoreGalanos** [comment]
Hi Anton and Chris,

I have this problem as well. Usually it comes down to my own bad practices in modelling and failing to really check my curves, which end up to surfaces, which end up to breps, which end up to zones.

Having good quality control while doing the model, early on if possible, is quite good. 

That said, any ways to automatically scrape off annoying little segments after intersecting just the curve outline of a layout plan?

Kind regards,
Theodore.

**Comment by mostaphaRoudsari** [comment]
All,
EnergyPlus has no check for models in 3d. It simply follows the hierarchy between the surfaces and the zone. @antonszilasi the example file that you shared is just poor modeling. Are you proposing that we should throw a warning in cases such as this? In that case we need to check every zone with every other zone which will add to the calculation time based on the number of the zones and the number of corners in each zone. We don't want to run this check for every honeybee simulation.

A good solution might be adding a component that does a quality check which one can use in case they like to pass the quality check of the geometry.

**Comment by chriswmackey** [comment]
I agree a quality check component is a good idea.  So people have the option of running intense calculations to tell them what they did wrong but they don't necessarily have to.

**Comment by antonszilasi** [comment]
gents that does sound like a good idea
